### ISSUE TITLE
Evaluate users of 3.12 tests and remove support for that compiler if impact is minimal

### ISSUE THREAD

**Issue by dsheets** [ocaml/ocaml-ci-scripts#82]
See #81 where 3.12 on OS X fails due to the well-known `-fno-defer-pop` issue when OS X made `gcc` an alias for `clang`.

**Comment by yomimono** [comment]
I recently cloned a bunch of repositories specified in the `dev-repo` field of released opam packages.  Of those, 717 had a .travis.yml .  Of those 717, 35 had .travis.yml files that included `3.12`; of those, only four referenced `ocaml-ci-scripts` or `ocaml-travis-ci-skeleton` (and two of them are duplicates):

```
yomimono@fuzzy:~/travis-bearing-repos$ find . -name .travis.yml | xargs grep -l  3.12|xargs grep -E "ocaml-travisci-skeleton|ocaml-ci-scripts"
./ocplib-endian/.travis.yml:install: wget https://raw.githubusercontent.com/ocaml/ocaml-ci-scripts/master/.travis-opam.sh
./dumpast/.travis.yml:install: wget https://raw.githubusercontent.com/ocaml/ocaml-travisci-skeleton/master/.travis-opam.sh
./subtype-refinement/.travis.yml:install: wget https://raw.githubusercontent.com/ocaml/ocaml-ci-scripts/master/.travis-opam.sh
./ocaml-dumpast/.travis.yml:install: wget https://raw.githubusercontent.com/ocaml/ocaml-travisci-skeleton/master/.travis-opam.sh
```

I think the impact would be minimal.

/cc @avsm

**Comment by yomimono** [comment]
`travis-opam` now uses jbuilder, which means it requires OCaml >= 4.02.
